packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -429,7 +429,11 @@ export class TdsDropdown {
 
   private setValueAttribute = () => {
     if (this.value?.toString() === '') this.value = null;
-    this.host.setAttribute('value', this.value?.map((val) => val).toString() ?? null);
+    if (this.value === null) {
+      this.host.removeAttribute('value');
+    } else {
+      this.host.setAttribute('value', this.value.map((val) => val).toString());
+    }
   };
```

Comment:
Should probably be: 

```jsx
  private setValueAttribute = () => {
    if (this.value === null || this.value?.toString() === '') {
      this.host.removeAttribute('value');
    } else {
      this.host.setAttribute('value', this.value.map((val) => val).toString());
    }
  };
```
or maybe even:
```jsx
  private setValueAttribute = () => {
    if (!this.value) {
      this.host.removeAttribute('value');
    } else {
      this.host.setAttribute('value', this.value.map((val) => val).toString());
    }
  };
```

Reply:
And `this.value = null;` that is currently being done on line 431 should not be done in this function. Since it's called `setValueAttribute` I think that is all it should do, not update component state.